File: Data Structures & Algorithms/is-anagram/submission-2.py
class Solution:
    def isAnagram(self, s: str, t: str) -> bool:
        map = {}
        for idx, val in enumerate(s): 
            map[val] = map.get(val, 0) + 1
        for idx, val in enumerate(t): 
            if val not in map:
                return False
            map[val] = map.get(val, 0) - 1 
            if map[val] < 0:
                return False
        return True 

File: Data Structures & Algorithms/combination-target-sum-ii/submission-2.js
class Solution {
    /**
     * @param {number[]} candidates
     * @param {number} target
     * @return {number[][]}
     */
    combinationSum2(candidates, target) {
        candidates.sort((a,b) => a - b);
        const result = []
        const dp = (idx, combination, sum) => {
            if(idx > candidates.length) return; 
            if(sum > target) return
            if(sum == target){
                result.push([...combination])
                return
            }

            for(let i = idx; i < candidates.length; i++){
                if(i > idx && candidates[i] == candidates[i - 1]) continue
                combination.push(candidates[i])
                dp(i + 1, [...combination], sum + candidates[i]);
                combination.pop()
            }
            
        }

        dp(0, [], 0)
        return result; 
    }
}
